resources/components/ImageCard.vue

```diff
@@ -10,13 +14,14 @@
 
 		<div class="wbmad-image-with-suggestions-tags">
 			<div class="wbmad-suggestion-group">
-				<ul>
-					<li v-for="(suggestion, index) in suggestions"
-						v-bind:key="index"
-						v-on:click="toggleConfirmed( suggestion )">
-						{{ suggestion }}
-					</li>
-				</ul>
+				<suggestion v-for="( suggestion, index ) in suggestions"
+					v-bind:key="index"
+					v-bind:text="suggestion.text"
+					v-bind:confirmed="suggestion.confirmed"
+					v-on:click="toggleConfirmed( suggestion )"
+				>
+					{{ suggestion.text }}
```

Comment:
`<suggestion>` components receive `text` both as a `prop` and as `<slot>` content. I assume this is to provide additional attributes for accessibility purposes? Ideally the user should only have to provide one of these things; it would be easy for someone to forget one or the other.

I think `<slot>` is best used for lower-level components or places where we need to pass in content with an arbitrary structure. Suggestion buttons are always just going to contain text content, so there is probably no need to use slots here. However, if you wanted to provide the content as a slot and wanted to re-use it within the component elsewhere (say for a `title` attribute somewhere), you can access slot content with `this.$children`. See the Tabs component for an example.

I still think that in this case it's best to just remove the slot and pass the content as props since we don't really need the additional flexibility here.

Reply:
Yeah, I was a bit torn on this one. For our current purposes just passing the text as a prop and getting rid of the slot would work. But I'm thinking of the new task to show aliases/descriptions. We wouldn't want to include all of that in the text prop (which is just used for the check icon title).

Maybe we could default to the text prop and only show the slot if it content is provided?

Reply:
I think the slot vs props thing comes down to reusability – will we need to re-use a given component in different ways / different places?

For the new "expand" feature, we will conditionally show extra data but that info (label, aliases, etc) will always be displayed in the same way. I think this is a good argument for using props here; using slots would imply that we don't always want this kind of consistency, and it places a little more burden on the developer (now you have to consult two different files to see what's going on).

I was saying earlier this week that we should prefer simple props to complex ones (rich objects with lots of attributes, etc), and I can see how having a cumbersome interface where we need to pass a ton of props to a given component could be a concern. I think in cases like this passing an object is okay as long as we're deliberate about it. For example, we could update the Suggestions model class to hold the new data we need and then pass that model object as a prop (maybe we can even validate for it). This could help alleviate the "where is all this data coming from" problem.

Reply:
What about a require prop called label that's used for the icon title/label as well as printed within the suggestion component, then an optional prop called content that could be additional content after the label? This is similar to what's done in OO.ui.MessageWidget.

Reply:
Sounds like a good approach.

Reply:
Per our slack conversation, I'm just going to include the text/label for now. We can worry about additional content later 😬